Weak delegates not correctly identified with Swift 4.1

### New Issue Checklist

- [X] Updated SwiftLint to the latest version
- [X] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

With Swift 4.1 that came out with Xcode 9.3 beta, the weak delegates don't seem to be identified correctly.

```bash
Linting Swift files in current working directory
Linting 'atest.swift' (1/1)
/Users/barden/projects/DTI/swiftlint-test/atest.swift:10:8: warning: Weak Delegate Violation: Delegates should be weak to avoid reference cycles. (weak_delegate)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)?
`0.24.2`
* Installation method used (Homebrew, CocoaPods, building from source, etc)?
`homebrew`

* Which Xcode version are you using (check `xcode-select -p`)?

/Applications/Xcode-9.3.app/Contents/Developer (9.3 beta from 24th Jan)

* Do you have a sample that shows the issue? 

```swift
protocol TestDelegate: class {

}

protocol TestProtocol {
  weak var delegate: TestDelegate? { get }
}

class Test {
  weak var delegate: TestDelegate
}
```


[Here](https://gist.github.com/dbarden/3727c106404b2015db765b86a8b5263a) there is the sourkitten structure output as well.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -23,6 +23,10 @@
   [Marcelo Fabri](https://github.com/marcelofabri)
   [#2017](https://github.com/realm/SwiftLint/issues/2017)
 
+* Fix several rules that use attributes when linting with Swift 4.1 toolchain.
+  [Marcelo Fabri](https://github.com/marcelofabri)
+  [#2019](https://github.com/realm/SwiftLint/issues/2019)
+
 ## 0.24.2: Dented Tumbler
 
 #### Breaking

--- a/Source/SwiftLintFramework/Extensions/Dictionary+SwiftLint.swift
+++ b/Source/SwiftLintFramework/Extensions/Dictionary+SwiftLint.swift
@@ -74,7 +74,8 @@ extension Dictionary where Key: ExpressibleByStringLiteral {
 
     var enclosedSwiftAttributes: [String] {
         let array = self["key.attributes"] as? [SourceKitRepresentable] ?? []
-        return array.flatMap { ($0 as? [String: String])?["key.attribute"] }
+        let dictionaries = array.flatMap { ($0 as? [String: SourceKitRepresentable]) }
+        return dictionaries.flatMap { $0["key.attribute"] as? String }
     }
 
     var substructure: [[String: SourceKitRepresentable]] {

